models/discordactions.js

```diff
@@ -46,6 +46,31 @@ const createNewRole = async (roleData) => {
   }
 };
 
+/**
+ * Soft deletes a group role by marking it as deleted in the database.
+ * This function updates the role document in Firestore, setting isDeleted to true
+ * and recording who deleted it and when.
+ *
+ * @param {string} groupId - The ID of the group role to be deleted
+ * @param {string} deletedBy - The ID of the user performing the deletion for logging purpose
+ * @returns {Promise<Object>} An object indicating whether the operation was successful
+ */
+const deleteGroupRole = async (groupId, deletedBy) => {
+  try {
+    const roleRef = admin.firestore().collection("discord-roles").doc(groupId);
+    await roleRef.update({
+      isDeleted: true,
+      deletedAt: admin.firestore.Timestamp.fromDate(new Date()),
```

Comment:
can you please confirm it that 
admin.firestore.Timestamp.fromDate(new Date())
is adding seconds and nanoseconds or not?

Reply:
Yes, the nanoseconds can't be seen in Firestore UI, but I confirm they are present in db. Please see the log in the screenshot attached.
<img width="1173" alt="Screenshot 2024-11-11 at 9 30 14 AM" src="https://github.com/user-attachments/assets/e7b6c3ef-d711-4362-8003-b3158aab02ee">